pkg/provision/storage/commonStorage.go

```diff
@@ -48,8 +48,12 @@ func (p *CommonStorageProvisioner) ProvisionStorage(podAdditions *v1alpha1.PodAd
 	}
 
 	if err := p.rewriteContainerVolumeMounts(workspace.Status.DevWorkspaceId, podAdditions, &workspace.Spec.Template); err != nil {
-		return err
+		return &ProvisioningError{
+			Err:     err,
```

Comment:
> I wonder why the message has duplicated text but if we can easily avoid it

The answer is in two places:
```go
		case *storage.ProvisioningError:
			return r.failWorkspace(workspace, fmt.Sprintf("Error provisioning storage: %s", storageErr), reqLogger, &reconcileStatus)

//...

// ProvisioningError represents an unrecoverable issue in provisioning storage for a DevWorkspace.
type ProvisioningError struct {
	// Err is the underlying error causing the problem. If nil, it is not included in the output of Error()
	Err error
	// Message is a user-friendly string explaining why the error occurred
	Message string
}

func (e *ProvisioningError) Error() string {
	if e.Err != nil {
		return fmt.Sprintf("%s: %s", e.Message, e.Err)
	}
	return e.Message
}
```

We set in the condition storageErr.Message, but fail workspace with storageErr.

We should not set err here or fail workspace with storageErr.

@amisevsk you're an author of ProvisioningError, what would be your suggestion here?

Reply:
The intention is that for `ProvisioningError`, `Message` is the important bit and `Err` is additional context, so that you can say something like
```go
return &ProvisioningError{
  Err: // Error from a library call,
  Message: "Failed to do X",
}
```

Message should not include the error, so either we should add context, e.g.
```go
return &ProvisioningError{
  Err:     err,
  Message: "Could not rewrite container volume mounts", // Something better than this of course
}
```

or  we should leave `Err` nil since the message is being consumed:
```go
return &ProvisioningError{
  Message: err.Error()
}
```

Reply:
The joys of Go error handling mean that you have to know what info is in the error so you don't repeat yourself :upside_down_face:

Reply:
Sorry! I noticed this but forgot to change it before i created the PR. I've taken @amisevsk example to add context